tests/components/cloud/test_tts.py

```diff
@@ -42,54 +54,138 @@ def test_schema() -> None:
     tts.PLATFORM_SCHEMA({"platform": "cloud"})
 
 
+@pytest.mark.parametrize(
+    ("engine_id", "platform_config"),
+    [
+        (
+            DOMAIN,
+            None,
+        ),
+        (
+            DOMAIN,
+            {
+                "platform": DOMAIN,
+                "service_name": "yaml",
+                "language": "fr-FR",
+                "gender": "female",
+            },
+        ),
+    ],
+)
 async def test_prefs_default_voice(
-    hass: HomeAssistant, cloud_with_prefs, cloud_prefs
+    hass: HomeAssistant,
+    cloud: MagicMock,
+    set_cloud_prefs: Callable[[dict[str, Any]], Coroutine[Any, Any, None]],
+    engine_id: str,
+    platform_config: dict[str, Any] | None,
 ) -> None:
     """Test cloud provider uses the preferences."""
-    assert cloud_prefs.tts_default_voice == ("en-US", "female")
+    assert await async_setup_component(hass, "homeassistant", {})
+    assert await async_setup_component(hass, TTS_DOMAIN, {TTS_DOMAIN: platform_config})
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+    await hass.async_block_till_done()
 
-    tts_info = {"platform_loaded": Mock()}
-    provider_pref = await tts.async_get_engine(
-        Mock(data={const.DOMAIN: cloud_with_prefs}), None, tts_info
-    )
-    provider_conf = await tts.async_get_engine(
-        Mock(data={const.DOMAIN: cloud_with_prefs}),
-        {"language": "fr-FR", "gender": "female"},
-        None,
-    )
+    assert cloud.client.prefs.tts_default_voice == ("en-US", "female")
 
-    assert provider_pref.default_language == "en-US"
-    assert provider_pref.default_options == {"gender": "female", "audio_output": "mp3"}
-    assert provider_conf.default_language == "fr-FR"
-    assert provider_conf.default_options == {"gender": "female", "audio_output": "mp3"}
+    on_start_callback = cloud.register_on_start.call_args[0][0]
+    await on_start_callback()
 
-    await cloud_prefs.async_update(tts_default_voice=("nl-NL", "male"))
-    await hass.async_block_till_done()
+    engine = get_engine_instance(hass, engine_id)
```

Comment:
There's no way to check the default language and options without accessing the entity directly currently.